### ISSUE TITLE
padding reveals an area without background color

### ISSUE THREAD

**Issue by tcurdt** [rivo/tview#833]
Unfortunately setting a padding reveals an area where I cannot set a background color.
Here the left and right padding is just white instead of black.

<img width="176" alt="Screenshot 2023-03-26 at 18 19 30" src="https://user-images.githubusercontent.com/13697/227789413-862eec2f-fc30-40a4-8eef-7e61cec8d1a5.png">

Why isn't the background color being used?

_Originally posted by @tcurdt in https://github.com/rivo/tview/issues/831#issuecomment-1484148541_

**Comment by rivo** [comment]
I tried to reproduce this but couldn't. Please post a small code snippet that reproduces this issue.

**Comment by digitallyserviced** [comment]
@tcurdt Sometimes this can be caused by certain containers not having their base `Box` primitive's `dontClear` value set right. 

I have had to hack a method to allow this to be modified so that I could ensure that existing, and any custom primitives I developed I could make sure that it cleared the background or not...

https://github.com/digitallyserviced/tview/blob/611a26b9bad195005135a6ce0215280c68d57903/box.go#L136

Specifically the part of the draw function that uses this when filling backgrounds.

```go
    background := def.Background(b.backgroundColor).Reverse(b.reverse)
	if !b.dontClear {
		for y := b.y; y < b.y+b.height; y++ {
			for x := b.x; x < b.x+b.width; x++ {
				screen.SetContent(x, y, ' ', nil, background)
			}
		}
	}

```

**Comment by tcurdt** [comment]
Here is a snippet from the code.

```
...
	layoutCol2 := tview.NewFlex().
		SetDirection(tview.FlexRow).
		AddItem(identificationTextview, 3, 0, false).
		AddItem(valueInput, 1, 0, false).
		AddItem(detailsTextview, 6, 0, false)
	layoutCol2.SetBorder(true)
	layoutCol2.SetBorderPadding(0, 0, 1, 1)

...

	layoutControl := tview.NewFlex().
		SetDirection(tview.FlexColumn).
		AddItem(layoutCol1, 50, 0, false).
		AddItem(layoutCol2, 23, 0, false).
		AddItem(layoutCol3, 0, 1, false).
		AddItem(layoutCol4, 40, 0, false)

...
```

Let me know if that already helps.
Otherwise I need to try and extract the problem somehow.

**Comment by digitallyserviced** [comment]
@tcurdt ya your code doesnt really help, because i am already pretty sure of the problem... you need to make dontClear available to your code because Flex is one of the primitives that defines `dontClear` as true

```go
func NewFlex() *Flex {
    f := &Flex{
        direction: FlexColumn,
    }
    f.Box = NewBox()
    f.Box.dontClear = true
    return f
}
```

I already provided the link to what needs to be added to make this modifiable from outside tview
https://github.com/digitallyserviced/tview/blob/611a26b9bad195005135a6ce0215280c68d57903/box.go#L136C1-L139C2

**Comment by digitallyserviced** [comment]
@tcurdt @rivo 
#860 should resolve this... it definitely should be available outside regardless... it is not just here it becomes an issue. This is why it is included in my fork.

**Comment by tcurdt** [comment]
@digitallyserviced So far it feels a little unintuitive having to bother with this though.

**Comment by digitallyserviced** [comment]
@tcurdt 
> So far it feels a little unintuitive having to bother with this though.

i agree but there are a lot of things in life that are unintitive or bothersome... and you're still alive

**Comment by rivo** [comment]
Can't reproduce this anymore. I take it's fixed now. If it's not, please provide a full runnable program to reproduce this in a new issue.
